res_handler_header_injector_test.go

```diff
@@ -0,0 +1,62 @@
+package main
+
+import (
+	"encoding/json"
+	"github.com/TykTechnologies/tyk/apidef"
+	"testing"
+
+	"github.com/TykTechnologies/tyk/test"
+	"github.com/TykTechnologies/tyk/user"
+)
+
+func TestResponseHeaderInjection(t *testing.T) {
+	ts := newTykTestServer()
+	defer ts.Close()
+	
+	buildAndLoadAPI(func(spec *APISpec) {
+        spec.UseKeylessAccess = true
+        spec.Proxy.ListenPath = "/"
+		spec.OrgID = "default"
+		updateAPIVersion(spec, "v1", func(v *apidef.VersionInfo) {
+			v.UseExtendedPaths = true
+			json.Unmarshal([]byte(`[
+				{
+					"delete_headers": [
+						"delete-this"
+					],
+					"add_headers": {
+						"add-this": "header"
+					},
+					"path": "/test-with-slash",
+					"method": "GET",
+					"act_on": false
+				},
+				{
+					"delete_headers": [
+						"delete-this"
+					],
+					"add_headers": {
+						"add-this": "header"
+					},
+					"path": "test-no-slash",
+					"method": "GET",
+					"act_on": false
+				}
+			]`), &v.ExtendedPaths.TransformResponseHeader)
+		})
+	})
```

Comment:
@buger Am I setting up the middleware correctly here? It seems like it's not getting triggered for either path

json is same as for my test API

Reply:
Check if json.Unmarshal returns error, maybe it will contain some useful debugging info